### DISCUSSION TITLE
Rename Nonce to BatchID

### DISCUSSION THREAD

**Discussion by evan-gray** [wormhole-foundation/wormhole#1968] [Protocol / feature design discussion]
As an integrator, nonce is an overloaded term which is not very self-descriptive. It is easy to misinterpret that this might be like a wallet nonce where the caller should be tracking and incrementing a counter, but this is essentially what the sequence number is. With #1679 it is proposed that this field be used to enable and group messages into batches. 

Should this field be renamed to something more obvious like `groupId` or `batchId`? Are there any concerns about renaming? As far as I understand the string is not used anywhere and the data would remain unchanged - this should merely be a cosmetic change for clarity’s sake.

**Comment by a5-pickle** [comment]
@gator-boi and I have been referring to these IDs as message IDs actually. But either of those terms sound good to me.

Speaking of being more obvious and clear, should terms like "VM" and "VAA" be swapped out for something more meaningful like "Wormhole message"? And "payload" which seems to be used everywhere be more descriptive depending on where the payload lives (e.g. for the generic message, `messagePayload`)?

Didn't mean to hijack this question with this other question. But I think there should be a discussion about more meaningful names in general. Then new integrators wouldn't have to learn Wormhole-ese to get things done.

**Comment by evan-gray** [comment]
@a5-pickle fwiw I just searched and the go code and protobuf refer to the `{chainId}/{emitterAddress}/{sequence}` as MessageId and already refer to the batchId as `{chainId}/{tx}` (though I'm not sure the latter will remain correct). Naming is hard.

https://github.com/wormhole-foundation/wormhole/blob/dev.v2/proto/publicrpc/v1/publicrpc.proto#L45-L59
